art/attacks/evasion/graphite/graphite_blackbox.py

```diff
@@ -0,0 +1,887 @@
+                obj_width=obj_width,
+                focal=focal,
+                clip_min=clip_min,
+                clip_max=clip_max,
+                mask=mask[i],
+                pts=pts,
+            )
+
+        y = to_categorical(y, self.estimator.nb_classes)  # type: ignore
+
+        # COMPUTE SUCCESS RATE
+        x_copy = np.zeros((x.shape[0], self.noise_size[1], self.noise_size[0], x.shape[3]))
+        x_adv_copy = np.zeros((x_adv.shape[0], self.noise_size[1], self.noise_size[0], x_adv.shape[3]))
+        for i in range(x_copy.shape[0]):
+            x_copy[i] = convert_to_network(x[i], self.net_size, clip_min, clip_max)
+            x_adv_copy[i] = convert_to_network(x_adv[i], self.net_size, clip_min, clip_max)
+
+        if self.estimator.channels_first:
+            x_copy = np.transpose(x_copy, (0, 3, 1, 2))
+            x_adv_copy = np.transpose(x_adv_copy, (0, 3, 1, 2))
+
+        logger.info(
+            "Success rate of GRAPHITE hard-label attack: %.2f%%",
+            100
+            * compute_success(
+                self.estimator,
+                x_copy.astype(np.float32),
+                y,
+                x_adv_copy.astype(np.float32),
+                self.targeted,
+                batch_size=self.batch_size,
+            ),
+        )
+
+        if self.estimator.channels_first:
+            x_adv = np.transpose(x_adv, (0, 3, 1, 2))
+
+        return x_adv
+
+    def _perturb(
+        self,
+        x: np.ndarray,
+        y: int,
+        x_tar: np.ndarray,
+        obj_width: float,
+        focal: float,
+        clip_min: float,
+        clip_max: float,
+        mask: Optional[np.ndarray] = None,
+        pts: Optional[np.ndarray] = None,
+    ) -> np.ndarray:
+        """
+        Internal attack function for one example.
+        :param x: An array with one original input to be attacked.
+        :param y: The target label.
+        :param x_tar: Initial array to act as an example target image.
+        :param obj_width: Estimated width of object in inches for perspective transform.
+        :param focal: Estimated focal length in ft for perspective transform.
+        :param clip_min: Minimum value of an example.
+        :param clip_max: Maximum value of an example.
+        :param mask: An array with a mask to be applied to the adversarial perturbations. Shape needs to be
+                     broadcastable to the shape of x. Any features for which the mask is zero will not be adversarially
+                     perturbed.
+        :param pts: Optional. A set of points that will set the crop size in the perspective transform.
+        :return: An adversarial example.
+        """
+        import cv2
+
+        x = (x.copy() - clip_min) / (clip_max - clip_min)
+        x_tar = (x_tar.copy() - clip_min) / (clip_max - clip_min)
+
+        if mask is None:
+            mask = np.ones((self.noise_size[1], self.noise_size[0], x.shape[2]))
+        mask = mask / np.max(mask)
```

Comment:
Renaming to the new variable name
```suggestion
        mask_array = mask_array / np.max(mask_array)
```